configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java

```diff
@@ -1169,33 +1177,40 @@ private void logConfigChangeActions(ConfigChangeActions actions, DeployLogger lo
+                .orElseThrow(() -> new NotFoundException("Unable to get logs for for " + applicationId + " (application not found)"));
+        Map<String, Integer> ports = logserverHostInfo(applicationVersions);
+        return ports.entrySet().stream()
+                .map(entry -> HttpURL.create(Scheme.http, DomainName.of(entry.getKey()), entry.getValue(), HttpURL.Path.parse("logs")))
+                .toList();
+    }
 
-        HostInfo logServerHostInfo = hostInfos.stream()
-                .filter(host -> host.getServices().stream()
+    // Finds hostname and port for logserver container for all models/versions
+    private Map<String, Integer> logserverHostInfo(ApplicationVersions applicationVersions) {
+        return applicationVersions.applications().stream()
+                .peek(app -> log.log(Level.FINE, "Finding logserver host and port for version " + app.getVespaVersion()))
+                .map(Application::getModel)
+                .flatMap(model -> model.getHosts().stream())
+                .filter(hostInfo -> hostInfo.getServices().stream()
                         .anyMatch(serviceInfo -> serviceInfo.getServiceType().equalsIgnoreCase("logserver")))
-                .findFirst().orElseThrow(() -> new IllegalArgumentException("Could not find host info for logserver"));
-
-        ServiceInfo logService = logServerHostInfo.getServices().stream()
-                                                  .filter(service -> LOGSERVER_CONTAINER.serviceName.equals(service.getServiceType()))
-                                                  .findFirst()
-                                                  .or(() -> logServerHostInfo.getServices().stream()
-                                                                             .filter(service -> CONTAINER.serviceName.equals(service.getServiceType()))
-                                                                             .findFirst())
-                                                  .orElseThrow(() -> new IllegalArgumentException("No container running on logserver host"));
-        int port = servicePort(logService);
-        return HttpURL.create(Scheme.http, DomainName.of(logServerHostInfo.getHostname()), port, HttpURL.Path.parse("logs"));
+                .map(hostInfo -> hostInfo.getServices().stream()
+                        .filter(service -> LOGSERVER_CONTAINER.serviceName.equals(service.getServiceType()))
+                        .findFirst()
+                        .or(() -> hostInfo.getServices().stream()
+                                .filter(service -> CONTAINER.serviceName.equals(service.getServiceType()))
+                                .findFirst())
+                        .orElseThrow(() -> new IllegalArgumentException("No container running on logserver host")))
+                .collect(Collectors.toMap(ServiceInfo::getHostName, this::servicePort, (a, b) -> a));
```

Comment:
I think you actually need `groupingBy` here?

Reply:
But then it's simpler to just return a list of pairs.

Reply:
Right, there _could_ be different hostnames, I guess. Will fix